pkg/proxy/proxy.go

```diff
@@ -47,22 +48,23 @@ const (
 )
 
 type Proxy struct {
-	app         application.Application
-	cl          client.Client
-	tokenParser *auth.TokenParser
-	spaceLister *handlers.SpaceLister
-	metrics     *metrics.ProxyMetrics
+	app            application.Application
+	cl             client.Client
+	tokenParser    *auth.TokenParser
+	spaceLister    *handlers.SpaceLister
+	metrics        *metrics.ProxyMetrics
+	getMembersFunc commoncluster.GetMemberClustersFunc
 }
 
-func NewProxy(app application.Application, proxyMetrics *metrics.ProxyMetrics) (*Proxy, error) {
+func NewProxy(app application.Application, proxyMetrics *metrics.ProxyMetrics, getMembersFunc commoncluster.GetMemberClustersFunc) (*Proxy, error) {
```

Comment:
is there a need for passing `getMembersFunc` or we could just default here to `cluster.GetMemberClusters`? As tests are in the same package, there we can just overwrite `p.getMembersFunc`

Reply:
TBH I would prefer to have tests in `proxy_test` package, so that functions are tested from a client perspective. 

Also I found it easier to follow if we use a composition pattern instead of overwriting properties in tests function, which makes it a bit more "hidden" IMO. But I don't have a strong opinion about this.